packages/dapp-connector/src/WalletApi/Cip30Wallet.ts

```diff
@@ -28,12 +31,16 @@ export const WalletApiMethodNames: WalletMethod[] = Object.values(CipMethodsMapp
  *
  * Only return the allowed API methods.
  */
-const wrapAndEnableApi = (walletApi: WalletApi, allowedApiMethods: WalletMethod[]): WalletApi => {
+const wrapAndEnableApi = (
+  walletApi: WalletApi,
+  { allowedApiMethods, enabledExtensions }: AllowedApiMethods
+): WalletApi => {
   const objectApi: WalletApi = {
     getActivePubStakeKeys: () => walletApi.getActivePubStakeKeys(),
     getBalance: () => walletApi.getBalance(),
     getChangeAddress: () => walletApi.getChangeAddress(),
     getCollateral: (params?: { amount?: Cbor }) => walletApi.getCollateral(params),
+    getExtensions: () => Promise.resolve(enabledExtensions),
```

Comment:
It's not truly necessary, but we should check if the wallet reports the extension as enabled:
```
// pseudo code below
enabledExtensions.filter(extension => walletApi.getExtensions().some(walletExt => walletExt === extension))
```

Reply:
Although, the supported extensions are actually static 😃 , so feel free to ignore the comment

Reply:
The `walletApi` which is created by `createWalletApi` in `cip30.ts` always returns all methods for all extensions so the `getExtensions` will also return all *supported* extensions, not the *enabled* ones (there is no `enable` here yet). 

What we need to return here in `Cip30Wallet` is the array of extensions passed to the `enable()` method which is also filtered in `#getAllowedApiMethods`, so we don't really need to do anything with `walletApi.getExtensions` if we assume `createWalletApi` is implemented correctly and returns all methods for all supported extensions.